### PR TITLE
Updated APIs to latest Consul version

### PR DESCRIPTION
**Pull request by sukidhar**
- Added Name based and ID based querying
- Updated APIs according to latest consul docs

### CODE REVIEW

**Comment by balena** [comment]
CC @enzoqtvf -> maybe you can help to merge this one.

File: lib/strategy/consul/endpoint.ex [hunk 1]
```diff
@@ -7,7 +7,7 @@ defmodule Cluster.Strategy.Consul.Endpoint do
 
   @callback build_url(URI.t(), Keyword.t()) :: URI.t()
 
-  @callback parse_response([map]) :: [ip]
+  @callback parse_response([map], list()) :: [ip]
```
Comment (szajbus) [inline lib/strategy/consul/endpoint.ex]: I think `Keyword.t()` fits better here.

```suggestion
  @callback parse_response([map], Keyword.t()) :: [ip]
```

**Review by szajbus — CHANGES_REQUESTED** [review]
It looks good to me.

@sukidhar can you also please update README to explain the new/changed config options: `service` and `expected`

File: lib/strategy/consul/agent.ex [hunk 2]
```diff
@@ -7,10 +7,42 @@ defmodule Cluster.Strategy.Consul.Agent do
   use Cluster.Strategy.Consul.Endpoint
 
   @impl true
-  def build_url(%URI{} = url, config) do
-    %{url | path: "/agent/service/#{config[:service_name]}"}
+  def build_url(%URI{query: query} = url, config) do
```
Comment (balena) [inline lib/strategy/consul/agent.ex:10]: This is a breaking change.

Before users were used to add `service_name`, and now they will need to use `:id` or `:service` instead. Very important to state this very clearly at the `README.md` and even instruct users how to migrate their code to the new version.

**Review by balena — CHANGES_REQUESTED** [review]
Please notice this is a breaking change and you would need to bump at least the minor version because of that. Also state at the `README.md` how to migrate.

File: lib/strategy/consul/agent.ex (hunk 2 above)
Comment (sukidhar) [reply lib/strategy/consul/agent.ex:10]: I have pushed changes to Readme file.

File: README.md [hunk 3]
```diff
@@ -64,16 +64,43 @@ config :libcluster,
         # Datacenter parameter while querying.
         dc: "dc1",
 
-        # The default service_name for children endpoints specifications.
-        service_name: "my-service",
+        # The default service for children endpoints specifications.
+        service: [name: "service_name"],
```
Comment (balena) [inline README.md:68]: Cool!

File: README.md [hunk 4]
```diff
@@ -87,3 +114,9 @@ end
 You can determine the latest version by running `mix hex.info libcluster_consul` in your shell, or by going to the `libcluster_consul` [page on Hex.pm](https://hex.pm/packages/libcluster_consul).
 
 Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_doc) and published on [HexDocs](https://hexdocs.pm). Once published, the docs can be found at [https://hexdocs.pm/libcluster_consul](https://hexdocs.pm/libcluster_consul).
+
+## Migration from 1.0.7 to 1.1.0
```
Comment (balena) [inline README.md:118]: Excellent!

**Review by balena — APPROVED** [review]
lgtm

**Comment by sukidhar** [comment]
> lgtm

Thank you so much.

**Review by szajbus — APPROVED** [review]
Perfect, thank you!

**Pull request merged**